### PR TITLE
feat(profiling): import sampled profiles

### PR DESCRIPTION
**Pull request by JonasBa**
Adds the ability to import sample profiles

### CODE REVIEW

File: tests/js/spec/utils/profiling/profile/eventedProfile.spec.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ describe('EventedProfile', () => {
         {type: 'C', at: 4, frame: 0},
       ],
       shared: {
-        frames: [{name: 'f0'}, {name: 'f1'}, {name: 'f2'}],
```
Comment (Zylphrex) [inline tests/js/spec/utils/profiling/profile/eventedProfile.spec.tsx:41]: Why did this test for `EventedProfile` change when you've only added `SampledProfile`?

File: static/app/utils/profiling/profile/sampledProfile.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        `Expected samples.length (${samples.length}) to equal weights.length (${weights.length})`
+      );
+    }
+
+    for (let i = 0; i < samples.length; i++) {
+      const stack = samples[i];
+      const weight = weights[i];
+
+      sampledProfile.appendSampleWithWeight(
+        stack.map(n => frameIndex[n]),
```
Comment (Zylphrex) [inline static/app/utils/profiling/profile/sampledProfile.tsx:33]: Can `frameIndex[n]` here be undefined? Should this guard against it?

File: static/app/utils/profiling/profile/sampledProfile.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      );
+    }
+
+    return sampledProfile.build();
+  }
+
+  appendSampleWithWeight(stack: Frame[], weight: number): void {
+    // Ignore samples with 0 weight
+    if (weight === 0) return;
+    if (isNaN(weight)) throw new Error('invalid weight');
```
Comment (Zylphrex) [inline static/app/utils/profiling/profile/sampledProfile.tsx:44]: In what scenarios can the weight be `NaN`?

File: tests/js/spec/utils/profiling/profile/eventedProfile.spec.tsx (hunk 1 above)
Comment (JonasBa) [reply tests/js/spec/utils/profiling/profile/eventedProfile.spec.tsx:41]: @Zylphrex it was just unused, I noticed it when I was adding the test for the sampled profile, so I removed it. The stack only contains 2 frames, this 3rd one was there by mistake

File: static/app/utils/profiling/profile/sampledProfile.tsx (hunk 3 above)
Comment (JonasBa) [reply static/app/utils/profiling/profile/sampledProfile.tsx:44]: @Zylphrex probably can’t happen, I think this was some old guard.

File: static/app/utils/profiling/profile/sampledProfile.tsx (hunk 2 above)
Comment (JonasBa) [reply static/app/utils/profiling/profile/sampledProfile.tsx:33]: It should not be, that would mean that the trace is corrupt, would be fine to throw here, I’ll add it

**Review by Zylphrex — APPROVED** [review]

**Pull request merged**